packages/qwik/src/optimizer/src/plugins/vite.ts

```diff
@@ -1179,7 +1179,8 @@ export function convertManifestToBundleGraph(manifest: QwikManifest): QwikBundle
         // external dependency
         continue;
       }
-      if (dep.isTask) {
+      // prevent importing all segments chunks
+      if (dep.hasSymbols) {
```

Comment:
Simpler implementation. The result is the same as checking if `dep.isEntry`. No need to check for `dep.isPlan` as @qwik-city-plan chunk doesn't contain symbols.

Reply:
Ok, in #7453 I just removed all of it and only disallow more than 10 dynamic deps because we're probably not going to guess right. That works fine too it seems.